### PR TITLE
feat(svm): query events script

### PR DESCRIPTION
**Pull request by md0x**
Changes proposed in this PR:
- Add script to read and print svm events in a human readable format.

### CODE REVIEW

**Review by chrismaree — APPROVED** [review]

File: src/svm/solanaProgramUtils.ts [hunk 1]
```diff
@@ -189,3 +191,69 @@ export async function subscribeToCpiEventsForProgram(
...
+  const depositId = deserialize(depositIdSchema, DepositId, Buffer.from(value));
+  return new BN(depositId.value).toString();
+}
+
+/**
+ * Stringifies a CPI event.
+ */
+export function stringifyCpiEvent(obj: any): any {
+  if (obj?.constructor?.toString()?.includes("PublicKey")) {
+    if (obj.toString().includes("111111111111")) {
```
Comment (Reinis-FRP) [inline src/svm/solanaProgramUtils.ts:235]: unlikely to have zeroes anywhere else than in the beginning, though, still better to restrict that only the first 12 bytes matching 0 are translated as EVM addresses

File: src/svm/solanaProgramUtils.ts [hunk 2]
```diff
@@ -189,3 +191,69 @@ export async function subscribeToCpiEventsForProgram(
...
+      // First 8 bytes are 0 for EVM addresses.
```
Comment (Reinis-FRP) [inline src/svm/solanaProgramUtils.ts:236]:
```suggestion
      // First 12 bytes are 0 for EVM addresses.
```

File: src/svm/solanaProgramUtils.ts (hunk 1 above)
Comment (md0x) [reply src/svm/solanaProgramUtils.ts]: 100%

File: scripts/svm/queryEvents.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+const idl = require("../../target/idl/svm_spoke.json");
+const program = new Program<SvmSpoke>(idl, provider);
+const programId = program.programId;
+console.log("SVM-Spoke Program ID:", programId.toString());
+
+const argvPromise = yargs(hideBin(process.argv)).option("eventName", {
+  type: "string",
+  demandOption: false,
+  describe: "Name of the event to query",
+  choices: [
```
[the remark this answers is not in the document]
Comment (md0x) [reply scripts/svm/queryEvents.ts:21]: The [EventCoder](https://github.com/across-protocol/contracts/blob/78e35850fc1afa1ffa83d26f116bf3678600313c/src/svm/solanaProgramUtils.ts#L84) returns them with the first letter in lowercase. I guess we could capitalize the events ourselves in the [processEventFromTx](https://github.com/across-protocol/contracts/blob/78e35850fc1afa1ffa83d26f116bf3678600313c/src/svm/solanaProgramUtils.ts#L55) function as a custom treatment, but we aren’t doing that. That’s why the event names should match those returned by the EventCoder.

**Review by Reinis-FRP — APPROVED** [review]
nice!

**Pull request merged**